src/onnx/parse_qlinearconcat.cpp

```diff
@@ -0,0 +1,134 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/onnx/padding.hpp>
+#include <migraphx/onnx/conv.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/onnx/broadcast_qdq.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/stringutils.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+/*
+ *********************************************************************************
+ *  Reference: see QLinearConcat in                                              *
+ *  https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md  *
+ *********************************************************************************
+
+com.microsoft.QLinearConcat
+Concatenate a list of tensors into a single tensor.All input tensors must have the same shape,
+except for the dimension size of the axis to concatenate on.
+
+Version
+This version of the operator has been available since version 1 of the 'com.microsoft' operator set.
+
+Attributes
+axis : int (required)
+Which axis to concat on
+
+Inputs (3 - ∞)
+Y_scale : TF
+Y's scale.
+
+Y_zero_point : T8
+Y's zero point.
+
+inputs (variadic, heterogeneous) : TV
+List of tensors/scale/zero_point for concatenation
+
+Outputs
+Y : T8
+Concatenated tensor
+
+Type Constraints
+T8 : tensor(uint8), tensor(int8)
+Constrain input and output types to 8 bit signed and unsigned tensors.
+
+TF : tensor(float)
+Constrain scale types to any float tensor type.
+
+TV : tensor(uint8), tensor(int8), tensor(float)
+Sequence of (Tensor, Scale, ZeroPoint) tuples. The type is sequence of (T8, TF, T8).
+*/
+
+struct parse_qlinearconcat : op_parser<parse_qlinearconcat>
+{
+    std::vector<op_desc> operators() const { return {{"QLinearConcat"}}; }
+
+    // basic type checking for QLinearConcat Operator
+    void check_inputs(const std::vector<instruction_ref>& args) const
+    {
+        auto args_size = args.size();
+        // at least 5 input tensors, can be 5, 8, 11 ...
+        if((args_size < 5) or ((args_size - 2) % 3 != 0))
+            MIGRAPHX_THROW("QLINEARCONCAT: missing inputs");
+
+        auto t0_shape = args[2]->get_shape();
+        auto t0_type  = t0_shape.type();
```

Comment:
Please also check for valid type allowed for t0_type.

Reply:
Ideally, checking for Y related values as well..

Reply:
What are the valid values allowed for `axis`, please?

Reply:
Added the checks for t0_type and y related values. Axis is an int value, I've parsed it as a float mistakenly, Iv1 fixed that as well.